pennylane/templates/subroutines/controlled_sequence.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
```

Comment:
```python
import pennylane as qml
from pennylane import numpy as np

@qml.prod
def U(thetas, wires):
  qml.RX(thetas[0], wires = 0)
  qml.RX(thetas[1], wires = 1)

dev = qml.device("default.qubit", wires = 4)

@qml.qnode(dev)
def circuit(thetas):
  qml.ControlledSequence(U(thetas, wires = [0,1]), control = [2,3])
  return qml.state()

thetas = np.array([1., 1.], requires_grad = True)
qml.grad(circuit)(thetas)
```

`DeviceError: Operator (Controlled(RX(<autograd.numpy.numpy_boxes.ArrayBox object at 0x792d57097480>, wires=[1]) @ RX(<autograd.numpy.numpy_boxes.ArrayBox object at 0x792d57094b80>, wires=[0]), control_wires=[2]))**2 not supported on default.qubit and does not provide a decomposition.`

Reply:
This also doesn't work with `QuantumPhaseEstimation` 😢 

I would consider adding functionality that isn't implemented for `QuantumPhaseEstimation` to be outside the scope of this PR

Reply:
I added this example as a test, and marked it as xfail. I'll look into this and the failing TF gradient test tomorrow, but I'll resolve those in a separate bug-fix PR, so that it doesn't block reviewing this one! 

The bug-fix PR can be merged next week if necessary :)

Reply:
sorry, it is possible that test is failing because the script is returning qml.state. You should return qml.expval for testing purposes

Reply:
smells like [this](https://github.com/PennyLaneAI/pennylane/blob/81667197ff9c8ec04bfa8a617a4517029f1a95e6/pennylane/devices/default_qubit_legacy.py#L233-L234) btw